invenio_rdm_records/records/jsonschemas/records/definitions-v2.0.0.json

```diff
@@ -24,9 +24,15 @@
       }
     ]
   },
-  "dateType": {
+  "date_type": {
+    "type": "object",
+    "additionalProperties": false,
     "description": "Type of the date.",
-    "type": "string"
+    "properties": {
+      "id": {
+        "type": "string"
```

Comment:
might have missed it in previous ones. But there is a root (records-resources) definition for `#identifier` (which in the end is a string) but is a good idea to keep things consistent. That way if we change how ids work we just need to change the definition.

e.g. https://github.com/inveniosoftware/invenio-rdm-records/blob/27f791992926223c398e4d7ea8f1c8033b6c7f65/invenio_rdm_records/records/jsonschemas/records/definitions-v2.0.0.json#L7